addons/sourcemod/scripting/gokz-replays/recording.sp

```diff
@@ -256,23 +265,49 @@ public Action SaveJump(Handle timer, DataPack dp)
 
 // =====[ PRIVATE ]=====
 
-static void StartRecording(int client)
+static void StartRunRecording(int client)
 {
     if (IsFakeClient(client))
     {
         return;
     }
     QueryClientConVar(client, "sensitivity", SensitivityCheck, client);
     QueryClientConVar(client, "m_yaw", MYAWCheck, client);
-    recordingIndexOnTimerStart[client] = GetRollingIndex(client);
     
     DiscardRecording(client);
     ResumeRecording(client);
+    
+    // Copy pre data
+    int index;
+    recordedRunData[client].Resize(preAndPostRunTickCount);
+    if (recordedRecentData[client].Length < preAndPostRunTickCount)
+    {
+    	index = recordingIndex[client] - preAndPostRunTickCount;
```

Comment:
You're missing a check here for if index is < 0. This could throw an out-of-bounds exception as it is.
This is why I had the Subtract method before, it does the check for you.